unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs

```diff
@@ -85,12 +89,69 @@ public static ServiceLocator CreateDefault()
 
             result.Register<ISocialApiBridge>(() =>
             {
-                ITransport TransportProvider() =>
-                    new WebSocketClientTransport("wss://rpc-social-service.decentraland.zone");
+                UniTask<ITransport> CreateWebSocketAndConnect(CancellationToken cancellationToken)
```

Comment:
As mentioned before, this is way too complex and the dependency is shallow. RPCSocialAPIBridge doesnt actually need a transport, it needs the client service, we should just provide that.

You have two ways of achieving that:
- Convert `Main.SetupServices` into an async method and same about `ServiceLocatorFactory.CreateDefault` so you can resolve your client service before injecting it into the RPCSocialApiBridge.

- You can pass RPCSocialApiBridge a UniTaskCompletionSource of the client service and await it.

I'd go with the first one, using a UniTaskCompletionSource will add lot of boilerplate in the already huge RPCSocialApiBridge class.

Reply:
Fixed it here: https://github.com/decentraland/unity-renderer/pull/5226/commits/a353e43b6be8681e79eafa5caa749012c430dab4
I think it is much better now.
Let me know if you find it OK.